### PR TITLE
Audible Alerts 2.0

### PR DESCRIPTION
**Pull request by shanelenagh**
I am (finally) submitting my pull request for the Audible Alerts 2.0 feature that I demo'ed a few days ago, incorporating just a few feedback items I got from that demo.  I am linking to a walkthrough video I did of the code changes below, as I think that is often the easiest way to demo the tech changes (short of an interactive walkthrough, which I am happy to do as well).  In summary, the following are the changes:
- Many more "callout" options: vertical speed, distance (with optional decimal precision), various traffic Id options (including none or full callsign), both colloquial and individual digit number options
- Configurable speaking speed (which does affect pitch); this is one that I added since the demo
- Ground traffic and low-speed filter options
- Ground traffic filter option for the location (visual) view as well; should have included this in the demo, as it was there, but forgot to
- Many, many optimizations to the CPU usage, mainly by employing smarter (non-busy) waiting and avoiding premature notification of waiting threads
- Factoring new features into small (private or final) methods, for JIT and AOT compilers to either inline or convert to native code, based on profile
- Preference screen cleanup

Here is the walkthrough video of how these were implemented, focusing on the core codebase changes and any items that changed since the demo:

https://drive.google.com/file/d/1FV8jNPj8-sky3ZKehIaHVuIDoSvvlXAU/view?usp=sharing

Addresses #371 (both visual and audible traffic alerts allow filtering of ground traffic)

### CODE REVIEW

File: app/src/main/java/com/ds/avare/storage/Preferences.java [hunk 1]
```diff
@@ -349,6 +349,10 @@ public float getAudibleTrafficAlertsMaxFrequency() {
         return Float.parseFloat(mPref.getString(mContext.getString(R.string.AudibleTrafficAlertsMaxFrequency), "15.0"));
     }
 
+    public float getAudibleTrafficAlertsMinSpeed() {
+        return Float.parseFloat(mPref.getString(mContext.getString(R.string.AudibleTrafficAlertsMinimumSpeed), "0.0"));
```
Comment (apps4av) [inline app/src/main/java/com/ds/avare/storage/Preferences.java:353]: Will this require a try catch, or is this by design always a number?

File: app/src/main/java/com/ds/avare/storage/Preferences.java [hunk 2]
```diff
@@ -361,6 +365,30 @@ public float getAudibleClosingInCriticalAlertRatio() {
         return (float) (Integer.parseInt(mPref.getString(mContext.getString(R.string.AudibleTrafficAlertsClosingCritalAlert), "0")) / 100.00);
     }
 
+    public float getAudibleTrafficAlertsAltitude() {
+        return Float.parseFloat(mPref.getString(mContext.getString(R.string.AudibleTrafficAlertsAltitude), "1000.0"));
```
Comment (apps4av) [inline app/src/main/java/com/ds/avare/storage/Preferences.java:369]: Do all/some of these float conversions require try catch?

**Review by apps4av — APPROVED** [review]

File: app/src/main/java/com/ds/avare/storage/Preferences.java (hunk 2 above)
Comment (shanelenagh) [reply app/src/main/java/com/ds/avare/storage/Preferences.java:369]: That is a good point.  Without them the feature just quietly fails (just tested with some alpha chars).  I will put that in now.  Was missing those on the previous version too.

File: app/src/main/java/com/ds/avare/storage/Preferences.java (hunk 1 above)
Comment (shanelenagh) [reply app/src/main/java/com/ds/avare/storage/Preferences.java]: Committed

**Pull request merged**